jar_test.go

```diff
@@ -1964,6 +1964,77 @@ func TestRemoveCookies(t *testing.T) {
 	}
 }
 
+func TestRemoveAll(t *testing.T) {
+	testRemoveAll(t, mustParseURL("https://www.apple.com"), "www.apple.com")
+}
+
+func TestRemoveAllRoot(t *testing.T) {
+	testRemoveAll(t, mustParseURL("https://www.apple.com"), "apple.com")
```

Comment:
I'm not convinced about this behaviour. Or the (currently untested) behaviour that removing cookies from foo.apple.com will also remove certs from bar.apple.com.